esrally/mechanic/supplier.py

```diff
@@ -49,14 +51,22 @@ def create(cfg, sources, distribution, car, plugins=None):
     template_renderer = TemplateRenderer(version=es_version, os_name=target_os, arch=target_arch)
 
     if build_needed:
-        raw_build_jdk = car.mandatory_var("build.jdk")
-        try:
-            build_jdk = int(raw_build_jdk)
-        except ValueError:
-            raise exceptions.SystemSetupError(f"Car config key [build.jdk] is invalid: [{raw_build_jdk}] (must be int)")
-
         es_src_dir = os.path.join(_src_dir(cfg), _config_value(src_config, "elasticsearch.src.subdir"))
-        builder = Builder(es_src_dir, build_jdk, paths.logs())
+
+        if source_build_method == "docker":
+            builder = DockerBuilder(src_dir=es_src_dir, log_dir=paths.logs(), client=docker.from_env())
+        else:
+            raw_build_jdk = car.mandatory_var("build.jdk")
+            try:
+                build_jdk = int(raw_build_jdk)
```

Comment:
I wonder, if we are casting to `int`, is there a need for catching a `ValueError` below?

Reply:
This is mainly to protect against things where someone sets `build.jdk = 18.0.2.1`:
```
>>> build_jdk = "18.0.2.1"
>>> int(build_jdk)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '18.0.2.1'
```